src/libs/actions/Policy.js

```diff
@@ -336,21 +384,28 @@ function invite(logins, welcomeNote, policyID) {
             // Save the personalDetails for the invited user in Onyx
             if (data.jsonCode === 200) {
                 Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, PersonalDetails.formatPersonalDetails(data.personalDetails));
-                Navigation.goBack();
+                const nextPolicy = _.clone(allPolicies[key]);
+                const nextEmployeeList = _.intersection(newEmployeeList, nextPolicy.pendingInvitations);
+                nextPolicy.employeeList = _.uniq([...nextPolicy.employeeList, ...nextEmployeeList]);
+                nextPolicy.pendingInvitations = _.difference(nextPolicy.pendingInvitations, nextEmployeeList);
+                Onyx.set(key, nextPolicy);
                 return;
             }
 
-            // If the operation failed, undo the optimistic addition
-            const policyDataWithoutLogin = _.clone(allPolicies[key]);
-            policyDataWithoutLogin.employeeList = _.without(allPolicies[key].employeeList, ...newEmployeeList);
+            const errorMessage = data.jsonCode === 402 ? `${Localize.translateLocal('workspace.invite.pleaseEnterValidLogin')}` : '';
+            throw new Error(errorMessage);
+        })
+        .catch((err) => {
+            // If the operationg failed undo the optimistic addition
+            const policyDataWithoutPending = _.clone(allPolicies[key]);
+            policyDataWithoutPending.pendingInvitations = _.without(policyDataWithoutPending.pendingInvitations, ...newEmployeeList);
 
-            // Show the user feedback that the addition failed
-            policyDataWithoutLogin.alertMessage = Localize.translateLocal('workspace.invite.genericFailureMessage');
-            if (data.jsonCode === 402) {
-                policyDataWithoutLogin.alertMessage += ` ${Localize.translateLocal('workspace.invite.pleaseEnterValidLogin')}`;
-            }
+            Onyx.set(key, policyDataWithoutPending);
 
-            Onyx.set(key, policyDataWithoutLogin);
+            const errorMessage = `${Localize.translateLocal('workspace.invite.genericFailureMessage')} ${err.message}`;
```

Comment:
The generic error messages should be used when we don't know the specific error. But we should not append the message from error to it.

So here, we should show the `pleaseEnterValidLogin` when the error code is 402 and `generic` for the rest of the errors. I am a little unsure of this pattern as it makes it complicated to differentiate between the user error(Our error) or HTTP error. 

For now, if you have a suggestion let me know. I will discuss this in the channel and suggest you some changes.